clang/lib/Basic/PointerAuthOptions.cpp

```diff
@@ -0,0 +1,34 @@
+//===- PointerAuthOptions.cpp -----------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "clang/Basic/PointerAuthOptions.h"
+
+using namespace clang;
+
+std::vector<PointerAuthSchema>
+clang::PointerAuthOptions::getAllUsedSchemas() const {
+  std::vector<PointerAuthSchema> Result;
+  Result.push_back(FunctionPointers);
+  Result.push_back(BlockInvocationFunctionPointers);
+  Result.push_back(BlockHelperFunctionPointers);
+  Result.push_back(BlockByrefHelperFunctionPointers);
+  Result.push_back(BlockDescriptorPointers);
+  Result.push_back(ObjCMethodListFunctionPointers);
+  Result.push_back(ObjCMethodListPointer);
+  Result.push_back(CXXVTablePointers);
+  Result.push_back(CXXTypeInfoVTablePointer);
+  Result.push_back(CXXVTTVTablePointers);
+  Result.push_back(CXXVirtualFunctionPointers);
+  Result.push_back(CXXVirtualVariadicFunctionPointers);
+  Result.push_back(CXXMemberFunctionPointers);
+  Result.push_back(ObjCIsaPointers);
+  Result.push_back(ObjCSuperPointers);
+  Result.push_back(InitFiniPointers);
+  assert(sizeof(Result[0]) * Result.size() + 4 == sizeof(PointerAuthOptions));
+  return Result;
```

Comment:
This definitely needs refactoring, but I didn't come up with a better way to enumerate all these `PointerAuthSchema`s yet.